confidence.py

```diff
@@ -81,6 +82,37 @@ def _split_keys(mapping, separator='.'):
             # use rest as the new key of value, recursively split that and update value
             value = _split_keys({rest: value}, separator)
 
+        if key in {'__abstractmethods__', '__class__', '__contains__',
+                   '__delattr__', '__dict__', '__dir__', '__doc__',
+                   '__eq__', '__format__', '__ge__', '__getattr__',
+                   '__getattribute__', '__getitem__', '__gt__', '__hash__',
+                   '__init__', '__init_subclass__', '__iter__', '__le__',
+                   '__len__', '__lt__', '__module__', '__ne__', '__new__',
+                   '__reduce__', '__reduce_ex__', '__repr__', '__reversed__',
+                   '__setattr__', '__sizeof__', '__slots__', '__str__',
+                   '__subclasshook__', '__weakref__', '_abc_cache',
+                   '_abc_negative_cache', '_abc_negative_cache_version',
+                   '_abc_registry', '_separator', '_source', 'get',
+                   'items', 'keys', 'values'}:
+            warnings.warn('The supplied configuration contains the key '
+                          '\'{reserved_key}\', which conflicts with '
+                          'methods used by mappings in Python. '
+                          'Accessing this attribute is not possible '
+                          'with the dot-separated notation, but only '
+                          'with the configuration\'s .get() '
+                          'method.'.format(reserved_key=key),
+                          UserWarning)
+
+        if key in {'clear', '__delitem__', 'setdefault', 'fromkeys',
+                   'popitem', 'copy', '__setitem__', 'pop', 'update'}:
```

Comment:
Not sure we need this; raising warnings for things that won't really break might be a tad over-protective. As long as we don't hard-code the set of attributes that *would* break things, new ones will automatically warn in the future.